[codelabs] update OTNS codelab to OTNS2

EskoDijk opened the pull request:
This updates the OTNS codelab to use the new OTNS2 simulator, which is currently not yet in the OpenThread project Github. The PR for OTNS2 is currently [here](https://github.com/openthread/ot-ns/pull/530).

It should be merged/accepted only after we have accepted OTNS2. This could be considered a draft PR, until that time. Some verification / fixes and checking required time may still be needed.

Vyrastas commented on site/en/codelabs/openthread-network-simulator/img/00_otns_web.png:
Can you also in this PR delete any old images that you've replaced?  No need to keep them.

EskoDijk replied:
Yes, good idea.

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -88,75 +83,53 @@ $ git clone https://github.com/openthread/ot-ns.git ./otns
-Now it's time to run OTNS...
+In this case, you need to add `$(go env GOPATH)/bin` to your `$PATH.`
 
+In case of other errors, a [Github issue](https://github.com/openthread/ot-ns/issues) can be created.
```
```suggestion
In case of other errors, a [GitHub issue](https://github.com/openthread/ot-ns/issues) can be created.
```

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -88,75 +83,53 @@ $ git clone https://github.com/openthread/ot-ns.git ./otns
-<img src="img/a0e05178d66929b1.png" alt="a0e05178d66929b1.png" width="624.00" />
+<img src="img/00_otns_web.png" alt="OTNS-Web window at start" width="624.00" />
 
 **If you can only see a blank page for OTNS-Web, chances are WebGL is not
 enabled in your browser. Please refer to**
```
Can you clean up this paragraph while you're at it, please.  Should just be a single bold around the whole thing, and needs a space between "to" and the link.

Vyrastas commented on site/en/codelabs/openthread-network-simulator/img/00_otns_web.png:
Can you add a black border around each image?  They way they are cut off is inconsistent when viewing it on the website (solid black line on one side, or on two).

EskoDijk replied:
Ok, so that should be black border of 1 pixel wide? I can do that.

Vyrastas replied:
Yeah 1 pixel is fine, just something to make them stand out better

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
````diff
@@ -545,16 +528,72 @@ Done
+7833912 00:00:06.543 [I] RouterTable---:     12 0x3000 - me - leader
+```
+
+The absolute simulation time in microseconds is shown on the left. The hh:mm:ss timestamp shows the OpenThread node's 
````
```suggestion
The absolute simulation time in microseconds is shown on the left. The `hh:mm:ss` timestamp shows the OpenThread node's 
```

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -545,16 +528,72 @@ Done
+
+<img src="img/13_wireshark.png" alt="Screenshot of OpenThread packet analysis in Wireshark" width="720" />
+
+To configure the decryption key, select in the menu Edit -> Preferences. Then in the preferences window, select 
+Protocols -> IEEE 802.15.4. Click the `Edit...` button next to "Decryption Keys". Click `+` to create a new entry 
+and enter the key `00112233445566778899aabbccddeeff` (32 characters) and select "Thread hash" in the "Key hash" field. 
+The "Decryption key index" can be left to `0`. Then click `OK`, and `OK` again. Now the OTNS PCAP file should 
+properly decrypt when loaded. 
```
Menus and buttons should be in bold.

```suggestion
To configure the decryption key, select in the menu **Edit -> Preferences**. Then in the preferences window, select 
**Protocols -> IEEE 802.15.4**. Click the **Edit...** button next to "Decryption Keys". Click **+** to create a new entry 
and enter the key `00112233445566778899aabbccddeeff` (32 characters) and select "Thread hash" in the "Key hash" field. 
The "Decryption key index" can be left to `0`. Then click **OK**, and **OK** again. Now the OTNS PCAP file should 
properly decrypt when loaded. 
```

Vyrastas reviewed (changes requested):
This all looks great, thanks for doing the update!  A few comments

EskoDijk commented:
@Vyrastas I've now made an update based on your review. Also added one more paragraph about timestamps correlation.

For merging this update we would probably have to wait until the OTNS2 PR is done (https://github.com/openthread/ot-ns/pull/530) ?

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -88,79 +83,57 @@ $ git clone https://github.com/openthread/ot-ns.git ./otns
-1. Run `which otns` to check if the `otns` executable is searchable in `$PATH.`
-2. If the `otns` command is not found, verify that you have added
-   `$(go env GOPATH)/bin` to `$PATH.`
+### Bootstrap and Install
```
```suggestion
### Bootstrap and install
```

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -88,79 +83,57 @@ $ git clone https://github.com/openthread/ot-ns.git ./otns
+In case of other errors, a [GitHub issue](https://github.com/openthread/ot-ns/issues) can be created.
 
-## Run OTNS
+## Run OTNS For the First Time
```
```suggestion
## Run OTNS for the first time
```

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -201,70 +197,80 @@ Duration: 05:00
+<img src="img/04_5n_add_router.png" alt="A Router is added, total 5 nodes" width="720" />
 
-<img src="img/420258bb92561146.png" alt="420258bb92561146.png" width="661.59" />
+Also click the FED, MED, SSED and BR buttons on the Action Bar to create these other types
```
```suggestion
Also click the FED, MED, SSED, and BR buttons on the Action Bar to create these other types
```

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -201,70 +197,80 @@ Duration: 05:00
 >
-> **Tip:** The blue line between Leader and Router indicates that they're linked.
+> **Tip:** A blue line between Leader, Router and FEDs indicates that they're linked.
+> Blue indicates it is not a Parent/Child link, but other: for example Router-to-Router.
```
```suggestion
> Blue indicates it is not a Parent/Child link, but something else, such as a Router-to-Router link.
```

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -545,16 +528,76 @@ Done
 
+An alternative way to exit node context is the `node 0` command.
+
+## Viewing Node Logs and Packet Captures
```
```suggestion
## Viewing node logs and packet captures
```

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -545,16 +528,76 @@ Done
+The absolute simulation time in microseconds is shown on the left. The `hh:mm:ss` timestamp shows the OpenThread node's 
+own log timestamp, which may differ from the absolute simulation time. 
+
+### Wireshark Packet Captures
```
```suggestion
### Wireshark packet captures
```

Vyrastas reviewed (changes requested):
Thanks for making all the image updates @EskoDijk !

Just some copyedit nits, otherwise looks good to me.  And yes we should wait for the other PR to be in before submitting this.

Have you had anyone else try out this updated version of the codelab?  I only reviewed from an editorial standpoint, I did not verify that it works as written.

EskoDijk commented:
Thanks, updated with the suggestions now.

> Have you had anyone else try out this updated version of the codelab? I only reviewed from an editorial standpoint, I did not verify that it works as written.

No, not yet. I could do a first pass myself to spot any issues, using [this rendering](https://github.com/EskoDijk/ot-docs/blob/pr-otns2-codelab/site/en/codelabs/openthread-network-simulator/index.lab.md). If we know any testing candidates let me know :) 

What I notice there is that some screenshots (ones that are wider) need to be enlarged, preferably to fill the entire width of the text-box/content-area in some cases.  

As a test, I added a commit using `width="95%"` (or similar) to set the width relative to the column width , for some pictures. The text is now much better readable. (In the real website, the column is slightly narrower though but comparable.)
So if that looks ok I'll adjust all the images to the right width.  For the wider screenshots, they need to fill 100% of column width while for the narrower screenshots it should be less.

Vyrastas commented on site/en/codelabs/openthread-network-simulator/index.lab.md:
```diff
@@ -418,78 +417,61 @@ OTNS enables users to move nodes easily through `OTNS-CLI` or `OTNS-Web`.
+button on the `Action Bar` to delete node 9:
 
-<img src="img/d4079cceea0105f0.png" alt="d4079cceea0105f0.png" width="642.12" />
+<img src="img/12_7n_del_br.png" alt="Border Router node 9 is deleted" width="642.12" />
```
I would make the width here consistent with the others, otherwise they are all fine

Vyrastas commented:
> As a test, I added a commit using `width="95%"` (or similar) to set the width relative to the column width , for some pictures. The text is now much better readable. (In the real website, the column is slightly narrower though but comparable.) So if that looks ok I'll adjust all the images to the right width. For the wider screenshots, they need to fill 100% of column width while for the narrower screenshots it should be less.

Everything looks fine on the openthread.io version (which I can't share externally), except for one image which I flagged (smaller width than the others).  Thanks for doing the extra work to make it look nice!

jwhui reviewed (approved):
Thanks!
